### PR TITLE
[User] Add food to cart

### PR DESCRIPTION
**Pull request by huyen99**
## Related Tickets
https://edu-redmine.sun-asterisk.vn/issues/39967

## WHAT
- Add food to cart from homepage and detail page
- Translate text in external javascript file

## HOW
- Jquery

## WHY
- N/A

## Evidence (Screenshot or Video)
235749071_364589291984962_1765708681872901654_n
236192711_447739976280924_9004053876527222950_n

### CODE REVIEW

**Comment by huyen99** [comment]
ready

File: main/views.py [hunk 1]
```diff
@@ -44,7 +51,13 @@ def register(request):
     
 def food_details(request, id):
     food = Food.objects.prefetch_related('review_set').annotate(avg_rating=Avg('review__rating')).filter(id=id).first()
+    in_cart = []
     
+    if request.user.is_authenticated:
```
Comment (huongnt-2545) [inline main/views.py:56]: tách thành function riêng rồi sử dụng, tránh lặp code ở nhiều nơi nha

File: main/views.py [hunk 2]
```diff
@@ -103,10 +117,33 @@ def reply(request, food_id, review_id):
...
+    bill = Bill.objects.prefetch_related('item_set').filter(user=request.user, status=status).first()
+    food = get_object_or_404(Food, id=request.POST.get('id'))
+    cart_items = bill.item_set.all()
+    action = ''
+
+    if cart_items.filter(food=food):
+        get_object_or_404(Item, food=food, bill=bill).delete()
+        action = 'remove'
+    else:
+        if food.discount: unit_price = float(food.price) * float(food.discount)
```
Comment (huongnt-2545) [inline main/views.py:140]:
```suggestion
        if food.discount:
          unit_price = float(food.price) * float(food.discount)
        else:
          unit_price = food.price
```

**Comment by huyen99** [comment]
ready

File: main/views.py [hunk 3]
```diff
@@ -11,6 +11,17 @@
...
 
+def get_cart(request):
+    bill, cart_items, in_cart = None, None, []
+    if request.user.is_authenticated:
+        status = get_object_or_404(Status, name='cart')
+        bill = Bill.objects.prefetch_related('item_set').filter(user=request.user, status=status).first()
+        cart_items = bill.item_set.all()
+        in_cart = [item.food for item in bill.item_set.all()]
+    
+    print(type(bill), type(cart_items), type(in_cart))
```
Comment (huongnt-2545) [inline main/views.py:22]: xóa debug

File: main/views.py [hunk 4]
```diff
@@ -102,11 +118,33 @@ def reply(request, food_id, review_id):
...
     }
     return render(request, 'cart/cart.html', context)
+
+@login_required
+def add_to_cart(request):
+    bill, cart_items, _ = get_cart(request)
+    food = get_object_or_404(Food, id=request.POST.get('id'))
+    action = ''
+
+    if cart_items.filter(food=food):
```
Comment (huongnt-2545) [inline main/views.py:135]: nên dùng `exists()` để check
```suggestion
    if cart_items.filter(food=food).exists():
```

**Comment by huyen99** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**